embedded-service/src/type_c/controller.rs

```diff
@@ -50,9 +55,28 @@ impl PortStatus {
         Self {
             available_source_contract: None,
             available_sink_contract: None,
-            connection_present: false,
-            debug_connection: false,
+            connection_state: None,
             dual_power: false,
+            alt_mode: AltMode::none(),
+            power_path: PowerPathStatus::none(),
+        }
+    }
+
+    /// Check if the port is connected
+    pub fn is_connected(&self) -> bool {
+        match self.connection_state {
+            Some(ConnectionState::Attached)
+            | Some(ConnectionState::DebugAccessory)
+            | Some(ConnectionState::AudioAccessory) => true,
+            _ => false,
+        }
+    }
```

Comment:
This is a `matches!`
```suggestion
    /// Check if the port is connected
    pub fn is_connected(&self) -> bool {
        matches!(self.connection_state, Some(ConnectionState::Attached) | Some(ConnectionState::DebugAccessory) | Some(ConnectionState::AudioAccessory))
    }
```

Same for `is_debug_accessory`

Reply:
Done.